### PR TITLE
Fix pep8

### LINKED ISSUE #400: flake8: Enable checking for F401 and others
Currently only select violation codes are checked and F401 is not one of them, causing unused imports in modules.

### PR DESCRIPTION
**Pull request by rosteen**
Building off of #407 , this will populate the ignore list and fix everything the other errors.

EDIT: Fix #400

### CODE REVIEW

File: setup.cfg [hunk 1]
```diff
@@ -70,7 +70,8 @@ addopts = --doctest-rst
 
 [flake8]
 max-line-length = 100
-select = E101,W191,W291,W292,W293,W391,E111,E112,E113,E502,E722,E901,E902
+# E402: Module level import not at top of file
+ignore = E402,W504,E226,F403,E126,E123,W503
```
Comment (pllim) [inline setup.cfg:74]: Having it alphanumerically ordered and have a list of comments to translate it makes it easier for future maintenance.

```suggestion
# E123: closing bracket does not match indentation of opening bracket's line
# E126: continuation line over-indented for hanging indent
# E226: missing whitespace around arithmetic operator
# F403: star import used; unable to detect undefined names
# W503: line break before binary operator 
# W504: line break after binary operator
ignore = E123,E126,E226,E402,F403,W503,W504
```

I would still advise against ignoring F403. It is safer to not use it unless you are managing namespace; and even then, use `# noqa` at the end of each affected line in the code.

File: setup.cfg [hunk 2]
```diff
@@ -70,7 +70,8 @@ addopts = --doctest-rst
 
 [flake8]
 max-line-length = 100
-select = E101,W191,W291,W292,W293,W391,E111,E112,E113,E502,E722,E901,E902
+# E402: Module level import not at top of file
```
Comment (pllim) [inline setup.cfg:73]: Should I selected 2 lines for the comment but I didn't. I moved this to my other code suggestion.
```suggestion
```

File: setup.cfg (hunk 1 above)
Comment (rosteen) [reply setup.cfg]: Thanks, committed this. We were planning to follow up separately with a review of our __init__ files for F403 and F401, but I may have time to get to it in this PR. If not I'll create an issue to do so.

File: jdaviz/configs/specviz/plugins/viewers.py [hunk 3]
```diff
@@ -347,7 +345,7 @@ def _plot_mask(self):
         if 'mask' in _data.components and self.display_mask:
 
             # If a mask was already plotted, remove it.
-            if hasattr(self,'mask_trace_pointer') and \
+            if hasattr(self, 'mask_trace_pointer') and \
                     self.mask_trace_pointer is not None:
```
Comment (pllim) [inline jdaviz/configs/specviz/plugins/viewers.py:349]: I am surprised it didn't complain about `\`.
```suggestion
            if (hasattr(self, 'mask_trace_pointer') and
                    self.mask_trace_pointer is not None):
```

File: jdaviz/configs/specviz/plugins/viewers.py [hunk 4]
```diff
@@ -388,7 +386,7 @@ def _plot_uncertainties(self):
             error = np.array(_data['uncertainty'].data)
 
             # If uncertainties were already plotted, remove them.
-            if hasattr(self,'error_trace_pointer') and \
+            if hasattr(self, 'error_trace_pointer') and \
                     self.error_trace_pointer is not None:
```
Comment (pllim) [inline jdaviz/configs/specviz/plugins/viewers.py:390]:
```suggestion
            if (hasattr(self, 'error_trace_pointer') and
                    self.error_trace_pointer is not None):
```

File: setup.cfg (hunk 2 above)
Comment (pllim) [reply setup.cfg:73]: Need to delete this now as suggested above. It is duplicated on L77.

**Review by pllim — COMMENTED** [review]
Nice clean up! Minor comments below.

**Review by javerbukh — APPROVED** [review]
All I can say is wow, nice work!

**Review by ibusko — APPROVED** [review]
Quite impressive!

**Pull request merged**